app/components/sidebars/main/channels_list/channel_item/index.js

```diff
@@ -33,7 +33,7 @@ function makeMapStateToProps() {
 
         if (channel.type === General.DM_CHANNEL) {
             if (ownProps.isSearchResult) {
-                isBot = channel.isBot;
+                isBot = channel.isBot ? channel.isBot : false;
```

Comment:
I think Boolean would be a more idiomatic way to do this. 
```suggestion
                isBot = Boolean(channel.isBot);
```